### PR TITLE
Replace gopass.GetPasswdMasked() by terminal.ReadPassword()

### PR DESCRIPTION
**Pull request by rithujohn191**
**What this PR does / why we need it**:
Replace `gopass.GetPasswdMasked()` used for reading passwords from the terminal with [`terminal.ReadPassword()`](https://godoc.org/golang.org/x/crypto/ssh/terminal#ReadPassword). This removes the `gopass` import.

**Special notes for your reviewer**:
Ran the following commands to update `godep` files:
```
./hack/godep-restore.sh -v
./hack/godep-save.sh
./hack/update-staging-godeps.sh
./hack/update-bazel.sh
```

/sig auth
/kind enhancement
/assign @ericchiang

```release-note
NONE
```

### CODE REVIEW

**Comment by ericchiang** [comment]
/ok-to-test
cc @sttts for more client-go dependency pruning

**Comment by liggitt** [comment]
Do we need to do the equivalent of a `if isTerminal(r.Fd())` check, or is that handled internally by the golang package as well?

**Comment by rithujohn191** [comment]
@liggitt  the `terminal.ReadPassword` method explicitly "gets" the terminal using the File Descriptor provided as shown [here](https://github.com/golang/crypto/blob/master/ssh/terminal/util.go#L98) and returns an error if it is unsuccessful. So I think we can get away without doing an explicit `isTerminal` check.

**Comment by ericchiang** [comment]
> ... the terminal.ReadPassword method explicitly "gets" the terminal using the File Descriptor provided as shown here and returns an error if it is unsuccessful. So I think we can get away without doing an explicit isTerminal check.

I think this behavior is different for every platform. Maybe it wouldn't hurt to throw in an `IsTerminal` check?

**Comment by rithujohn191** [comment]
FYI the failed test is due to https://github.com/kubernetes/kubernetes/issues/61893

**Comment by rithujohn191** [comment]
Added a `terminal.IsTerminal` check before reading the password

**Comment by liggitt** [comment]
/assign @thockin 
for removal approval

**Comment by ericchiang** [comment]
(as a dep-approver @sttts can also approve this PR)

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"ImportPath": "k8s.io/kubernetes",
-	"GoVersion": "go1.9",
+	"GoVersion": "go1.10",
```
Comment (thockin) [inline Godeps/Godeps.json]: we don't officially support go 1.10 yet - can you revert this one line?

Comment (rithujohn191) [reply Godeps/Godeps.json]: Done

**Pull request merged**